src/matrices/clement.jl

```diff
@@ -15,29 +15,43 @@ diagonal entries. If k = 1, the matrix is symmetric.
 for test purposes, SIAM Rev., 1 (1959), pp. 50-52,
 https://doi.org/10.1137/1001006.
 """
-abstract type Clement{T<:Number} <: AbstractMatrix{T} end
+struct Clement{T<:Number} <: AbstractMatrix{T} 
+    n::Integer
+    M::AbstractMatrix{T}
+
+    function Clement{T}(n::Integer, k::Integer) where {T}
+        # 1x1 case
+        if n == 1
+            return zeros(T, 1, 1)
+        end
+    
+        # create the matrix
+        x = T[n-1:-1:1;]
+        z = T[1:n-1;]
+        if k == 0
+            return new{T}(n, Tridiagonal(x, zeros(T, n), z))
+        else
+            y = sqrt.(x .* z)
+            return new{T}(n, SymTridiagonal(zeros(T, n), y))
+        end
+    end    
+end
```

Comment:
The reason for doing a separate construct function is to ensure that the return matrix is in `Tridiagonal` or `SymTridiagonal` to gain performance advantage when used. If we do that, we might need to update the docs of this type.

Reply:
I see. One issue with having the constructor for `Clement` return an object of a different type is that we cannot write any functions for Clement specifically -- we can only rely on those for `Tridiagonal` and `SymTridiagonal`. For example, the eigenvalues, inverse, and determinant are known.